Detached HEAD state in GitHub Actions on release.published

When running `pnpm publish` in a GitHub Actions workflow triggered by the `release.published` event, the following error occurs:

```bash
 ERR_PNPM_GIT_UNKNOWN_BRANCH  The Git HEAD may not be attached to any branch, but your "publish-branch" is set to "master|main".
```

This error occurs because the Git HEAD is detached during the release process. In GitHub Actions, when the release event occurs, the workflow typically checks out a specific commit or tag, which places the repository in a detached HEAD state.

As a result, `pnpm` fails to publish the package, as it expects the HEAD to be attached to a branch (e.g., `master` or `main`) for publishing, causing the process to fail.

--- a/.github/workflows/publish.yml
+++ b/.github/workflows/publish.yml
@@ -34,6 +34,6 @@ jobs:
         run: pnpm build
 
       - name: Publish to NPM
-        run: pnpm publish
+        run: pnpm publish --no-git-checks
         env:
           NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}

